pkg/tekton/pipeline_info.go

```diff
@@ -193,14 +195,13 @@ func CreatePipelineRunInfo(prName string, podList *corev1.PodList, ps *v1.Pipeli
 					gitURL = value
 				case "-revision":
 					if shaRegexp.MatchString(value) {
-						lastCommitSha = value
+						shaFromGitInit = value
```

Comment:
```
if strings.HasPrefix(container.Name, "build-step-git-source") || strings.HasPrefix(container.Name, "step-git-source")
```

I find this a bit worrying, that one needs to match against container names and pipeline steps. That seems fragile, or am I missing something here?

Reply:
So first we try to get git informtion from the actual checkout and then override with env variables if set, right?

Reply:
I can't remember all of the history here, but I think this logic basically predated us realizing all the information we need regarding the SHA etc is actually in Prow/Lighthouse-provided env vars. So I'm opting not to mess with this janky logic, but I am making it the fallback, rather than the current behavior where we don't set a bunch of other things if we found a SHA here.

This change specifically fixes the problem of the `lastCommitSha` for batch builds being the master SHA, not the first-PR-in-the-batch HEAD SHA, because batch builds always have call the git source step for master (and we don't actually care what ref the git source step is checking out for any PR builds, in practice, because our own bespoke git merge step always checks out master and merges onto it regardless).

Reply:
maybe one day I will fully understand what you just wrote ;-) It's the batching I don't fully grasp atm. I get it conceptually, but not sure how it actually works and how it keeps state for everything.

Reply:
Yeah, I admit that I don't entirely understand the batch logic in `kube/activity.go` - something else that probably needs to be rewritten. I'm also still not yet sure if the SHA this PR adds will actually suffice for Lighthouse and batch. We'll see.